server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -1917,6 +1919,26 @@ private void internalAlignedCreateTimeseries(
         encodings,
         TSFileDescriptor.getInstance().getConfig().getCompressor());
   }
+
+  /**
+   * Before to insert, set using template if template accessed. Better performance if refactored
+   * with getDeviceNodeWithAutoCreate
+   */
+  private void activateTemplateBeforeAccess(PartialPath deviceId, String[] measurements)
+      throws MetadataException {
+    StringBuilder builder = new StringBuilder(deviceId.getFullPath());
+    if (builder.length() != 0) {
+      builder.append(TsFileConstant.PATH_SEPARATOR);
+    }
+    for (String measurement : measurements) {
+      builder.append(measurement);
+      IMNode mountedNode = mtree.getTemplateMountedNode(new PartialPath(builder.toString()));
+      if (mountedNode != null && !mountedNode.isUseTemplate()) {
+        setUsingSchemaTemplate(mountedNode);
+      }
+      builder.delete(builder.length() - measurement.length(), builder.length());
+    }
+  }
```

Comment:
If the mountedNode has not been created on MTree, how could you activate it?

Reply:
Thanks for you insight!
Now, I will get the template mounted node index, rather than IMNode, so that auto create process methods are decoupled from mounted node detecting.